src/database/operations/create/createDataProvider.ts

```diff
@@ -0,0 +1,25 @@
+import { db } from '../../db';
+import type {
+	JsonResultSet,
+	DataProvider,
+	WritableDataProvider,
+} from '../../../types';
+
+export const createDataProvider = async (
+	createValues: WritableDataProvider,
+): Promise<DataProvider> => {
+	const { name } = createValues;
+	const result = await db.sql<JsonResultSet<DataProvider>>(
+		'dataProviders.insertOne',
+		{
+			name,
+		},
+	);
+	const { object } = result.rows[0] ?? {};
+	if (object === undefined) {
+		throw new Error(
+			'The entity creation did not appear to fail, but no data was returned by the operation.',
```

Comment:
That's a good point -- I imagine this would involve changing the signature for these functions to take a transaction context as a parameter, which is a better function design anyway.

I'll open up an issue to capture this improvement.

Reply:
Issue created: https://github.com/PhilanthropyDataCommons/service/issues/1147